### PR TITLE
Sort function

### PR DESCRIPTION
**Pull request by mirzaees**
replacing sort_function with natsort module because:
1 - It seems that it is not written properly as it does not return a sorted list and there is no return
2 - instead of importing 're' module and writing 2 functions for this purpose, it is more convenient to import only natsort module which is more general

### CODE REVIEW

File: rinsar/_process_utilities.py [hunk 1]
```diff
@@ -418,24 +418,25 @@ def file_len(fname):
 ##########################################################################
 
 
-def alphanum_key(s):
-    """ Turn a string into a list of string and number chunks.
-        "z23a" -> ["z", 23, "a"]
-    """
-    return [ int(c) if c.isdigit() else c for c in re.split('([0-9]+)', s) ]
+#def alphanum_key(s):
```
Comment (gravelcycles) [inline rinsar/_process_utilities.py]: Could you delete these commented-out functions? A good principle is that no commented-out code should be committed

File: rinsar/_process_utilities.py [hunk 2]
```diff
@@ -418,24 +418,25 @@ def file_len(fname):
...
+#    l.sort(key=alphanum_key)
 
 ##########################################################################
 
 def remove_zero_size_or_length_error_files(directory):
     """Removes files with zero size or zero length (*.e files in run_files)."""
     
     error_files  = glob.glob(directory + '/*.e')
-    sort_nicely(error_files)
+    #sort_nicely(error_files)
```
Comment (gravelcycles) [inline rinsar/_process_utilities.py]: Can you remove these commented-out lines too?

File: rinsar/_process_utilities.py [hunk 3]
```diff
@@ -19,7 +19,7 @@
 from rinsar.rsmas_logging import RsmasLogger, loglevel
```
Comment (gravelcycles) [inline rinsar/_process_utilities.py:19]: Can you remove the `re` import from `_process_utilities.py` as it is no longer needed?

**Review by gravelcycles — CHANGES_REQUESTED** [review]
`natsort` looks like a good module to use here. A few small comments and one big question: Do we have a `requirements.txt` file for Python modules we use/require to run `rsmas_insar`? If so, could you add `natsort` to it? If not, time for a new issue!!

**Comment by mirzaees** [comment]
Thank you, I fixed those lines. 
we have a requirements.txt file but it is not in a good way. it is a long list with all versions included which every few days when a package or conda is updated, some of them may disappear and conflicts raise. 
Also a long list requires a long time for conda to find matching packages. do you know a good way to create requirement.txt file list which only contains main necessary packages without version specified, similar to pysar? I don't know which packages are independent and which are not

**Review by gravelcycles — APPROVED** [review]

**Comment by gravelcycles** [comment]
@mirzaees good point. I know Falk brought up similar issues. Can you create an issue around cleaning up our `requirements.txt` file?

**Pull request merged**